bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/azure/AzureReplicatedBagVerifier.scala

```diff
@@ -0,0 +1,44 @@
+package uk.ac.wellcome.platform.archive.bagverifier.services.azure
+
+import com.amazonaws.services.s3.AmazonS3
+import com.azure.storage.blob.BlobServiceClient
+import uk.ac.wellcome.platform.archive.bagverifier.fixity.FixityChecker
+import uk.ac.wellcome.platform.archive.bagverifier.services.ReplicatedBagVerifier
+import uk.ac.wellcome.platform.archive.bagverifier.storage.Resolvable
+import uk.ac.wellcome.platform.archive.bagverifier.storage.azure.AzureResolvable
+import uk.ac.wellcome.platform.archive.common.bagit.services.BagReader
+import uk.ac.wellcome.platform.archive.common.bagit.services.azure.AzureBagReader
+import uk.ac.wellcome.storage.azure.{AzureBlobLocation, AzureBlobLocationPrefix}
+import uk.ac.wellcome.storage.listing.Listing
+import uk.ac.wellcome.storage.listing.azure.AzureBlobLocationListing
+import uk.ac.wellcome.storage.s3.S3ObjectLocation
+import uk.ac.wellcome.storage.store.StreamStore
+import uk.ac.wellcome.storage.store.azure.AzureStreamStore
+import uk.ac.wellcome.storage.store.s3.S3StreamStore
+
+class AzureReplicatedBagVerifier(val bucket: String)(
+  implicit val s3Client: AmazonS3,
+  implicit val blobClient: BlobServiceClient
```

Comment:
nit: is there a reason these are both `val`?

Reply:
So now I look at this, I wonder – maybe you expect an implicit `AmazonS3` in ReplicatedBagVerifier, and declare the `srcStreamStore` there?

Reply:
Aha, I see now, I hadn't read all the way down – so can you delete the `srcStreamStore` here?

Reply:
Good spot, they don't need to be both val

Reply:
I think maybe it does need to be val after all, but the compiler will tell you for sure.

Reply:
the s3Client does, but the blobClient doesn't. Anyway I've also removed srcStreamStore, thanks